app/scripts/lib/stores/read-only-network-store.ts

```diff
@@ -0,0 +1,91 @@
+import log from 'loglevel';
+import getFetchWithTimeout from '../../../../shared/modules/fetch-with-timeout';
+import {
+  type IntermediaryStateType,
+  BaseStore,
+  MetaMaskStorageStructure,
+} from './base-store';
+
+const fetchWithTimeout = getFetchWithTimeout();
+
+const FIXTURE_SERVER_HOST = 'localhost';
+const FIXTURE_SERVER_PORT = 12345;
+const FIXTURE_SERVER_URL = `http://${FIXTURE_SERVER_HOST}:${FIXTURE_SERVER_PORT}/state.json`;
+
+/**
+ * A read-only network-based storage wrapper
+ */
+export default class ReadOnlyNetworkStore extends BaseStore {
+  #initialized: boolean;
+
+  #initializing?: Promise<void>;
+
+  #state: IntermediaryStateType | null;
+
+  constructor() {
+    super();
+    this.#initialized = false;
+    this.#initializing = this.#init();
+    this.#state = null;
```

Comment:
Nit: Properties can be initialized inline

```suggestion
  #initialized: boolean = false;

  #initializing?: Promise<void>;

  #state: IntermediaryStateType | null = null;

  constructor() {
    super();
    this.#initializing = this.#init();
```